spec/models/user_spec.rb

```diff
@@ -3,9 +3,43 @@
 require 'rails_helper'
 
 RSpec.describe User, type: :model do
-  let(:user) { Fabricate(:user) }
+  describe '.from_omniauth' do
+    context 'given an existing user is found' do
+      it 'returns the existing user' do
+        auth = OmniAuth::AuthHash.new(Faker::Omniauth.google)
+        user = Fabricate(:user,
+                         provider: auth['provider'],
+                         uid: auth['uid'],
+                         email: auth['info']['email'])
 
-  it 'has email' do
-    expect(user.email).not_to be_nil
+        expect(described_class.from_omniauth(auth)).to eq(user)
+      end
+    end
+
+    context 'given a new user' do
+      it 'creates a new user with the provided attributes' do
+        auth = OmniAuth::AuthHash.new(Faker::Omniauth.google)
+
+        user = described_class.from_omniauth(auth)
```

Comment:
```suggestion
        expect do
          described_class.from_omniauth(auth)
        end.to change(described_class, :count).by(1)
```

Based on the other test. So in case this test is failing, it can suggest the next test has a false positive :)